### ISSUE TITLE
Make the componentIndex optional

### ISSUE THREAD

**Issue by Inkdpixels** [reduct/assembler#11]
Sometimes you won't need the componentIndex at all initially, for example, if you fully rely on the addComponent method. In these cases, this extra empty object is unnecessary boilerplate as @akoenig correctly stated.

**Comment by akoenig** [comment]
I considered it already in the next (possible) API design. Stay tuned :)

**Comment by Inkdpixels** [comment]
So exciting! :+1: 

![Tense](http://replygif.net/i/104.gif)

**Comment by akoenig** [comment]
:D

The new API surface draft is implemented in cee71afa446696cf656eb35c805fd8deaac5132b

``` js
let assembler from 'assembler';

import MyComponent from 'MyComponent';

let app = assembler();

app.register(MyComponent);

app.boot();
```

I do not think that we need an index anymore (when we follow this approach).
